paper.md

```diff
@@ -0,0 +1,159 @@
+---
+title: 'A Python Project Template for Healthy Scientific Software'
+tags:
+  - Python
+  - template
+authors:
+  - name: Drew Oldag
+    orcid: 0000-0001-6984-8411
+    equal-contrib: true
+    affiliation: "1, 2"
+  - name: Melissa DeLucchi
+    orcid: 0000-0002-1074-2900
+    equal-contrib: true
+    affiliation: "1, 3"
+  - name: Wilson Beebe
+    orcid: 0009-0003-1791-8707
+    affiliation: "1, 2"
+  - name: Doug Branton
+    orcid: 0009-0009-7822-7110
+    affiliation: "1, 2"
+  - name: Sandro Campos
+    orcid: 0009-0007-9870-9032
+    affiliation: "1, 3"
+  - name: Carl Christofferson
+    orcid: 0009-0009-1164-3494
+    affiliation: "1, 2"
+  - name: Andrew Connolly
+    orcid: 0000-0001-5576-8189
+    affiliation: "1, 2"
+  - name: Jeremy Kubica
+    orcid: 0009-0009-2281-7031
+    affiliation: "1, 3"
+  - name: Olivia Lynn
+    orcid: 0000-0001-5028-146X
+    affiliation: "1, 3"
+  - name: Konstantin Malanchev
+    orcid: 0000-0001-7179-7406
+    affiliation: "1, 3"
+  - name: Alex Malz
+    orcid: 0000-0002-8676-1622
+    affiliation: "1, 3"
+  - name: Rachel Mandelbaum
+    orcid: 0000-0003-2271-1527
+    affiliation: "1, 3"
+  - name: Sean McGuire
+    orcid: 0009-0005-8764-2608
+    affiliation: "1, 3"
+  - name: Chris Wenneman
+    orcid: 0009-0002-6218-5007
+    affiliation: "1, 2"
+affiliations:
+  - name: LINCC-Frameworks
+    index: 1
+  - name: DiRAC Institute and the Department of Astronomy, University of Washington, 3910 15th Ave NE, Seattle, WA 98195, USA
+    index: 2
+  - name: The McWilliams Center for Cosmology & Astrophysics, Department of Physics, Carnegie Mellon University, Pittsburgh, PA 15213, USA
+    index: 3
+date: 28 February 2024
+bibliography: paper.bib
+---
+
+# Summary
+
+The creation of healthy code is vital for its successful long term use in scientific research. To maximize impact throughout the community, software packages must be accurate, usable, and maintainable. Here we discuss several engineering processes that are important for developing healthy software. Unfortunately these processes often require configuration leading to short-term overhead for new projects. We introduce the LINCC-Frameworks Python Project Template, a configurable template designed for scientific software projects that greatly simplifies adopting such practices by automating the setup and configuration of important code health tools.
```

Comment:
I am wondering what is meant by "accurate" (in "software packages must be accurate").  Is this a statement about numerical robustness of results, ...?  It read a bit oddly to me.  Possible alternate word that might resonate with science users: "robust"?

Reply:
I interpreted "accurate" as more closely related to "correct". Meaning that as the user makes changes, they can be confident that the unit tests will prevent breaking the code in such a way that it would produce incorrect output. 

I tend to think of "robust" as meaning "resilient" and will protect the user in case of odd inputs and provide meaningful error messages if the code can't recover on its own. However, if "robust" has a different connotation in the context of scientific software, I would be all for swapping in "robust" :) 

Do you have a strong opinion "robust" vs. "correct" vs. something else?

Reply:
Going with "robust" for now.

Reply:
Updated

Reply:
I was just going to suggest either keeping "accurate" or going with "correct." The thinking was that it is important to make sure the code stays correct as changes are made. We've come across a few errors in various projects that could have been addressed by better testing.

Reply:
I'm also happy to go with "correct" :)

Reply:
Hmm.  So let me explain the origin of my confusion in case helpful (sorry for not saying more before).  As a scientist, I know that in general most of my analysis algorithms that I implement have some approximations or limitations, and are not formally correct.  What I care about is that once I have implemented it with all of its warts, that the code accurately reflect my intentions.  I wouldn't say that such code is necessarily "accurate" or "correct" as it feels like there is an absoluteness to that statement that is a mismatch to the situation... do you see what you mean?  And maybe I'm just a bit of a weirdo in this regard, so it's OK if you want to ignore this feedback and move on.

Reply:
I believe I understand what you're saying @rmandelb, and it makes a lot of sense. To @jeremykubica point, the thought that we're trying to convey here with one word (that we haven't quite figured out yet) is "confident that there will be no regressions as changes are made". 

So perhaps instead of "correct", or "accurate", we could say, "stable" or "consistent"? Something like:
"software packages must be usable, maintainable, and produce consistent results"

Reply:
I would be happy with that, thank you!  I'm OK if it's wrong for reasons I know/accept when I chose to implement a a particular algorithm.  It's the consistency with my intentions that matters.